### ISSUE TITLE
Two classes are generated when referencing within a schema

### ISSUE THREAD

**Issue by jshum2479** [joelittlejohn/jsonschema2pojo#623]
I have a schema like this 

{
    "id": "Test",
    "$schema": "http://json-schema.org/draft-04/schema#",
    "type": "object",
    "properties": {

    "myobjectas": {
        "type": "array",
        "items": {
			"name": "myobjecta",
			"type": "object",
			"properties": {
				"value": {
					"type": "string"
				}
			}
		}
    }
	,

    "myobjectbs": {
        "type": "array",
        "items": {
			"name": "myobjectb",
			"type": "object",
			"properties": {
				"value": {
					"type": "string"
				}
			}
		}
    },

    "AnotherObjects" : {
    	"type" : "object",
    	"properties" : {
    		"attribute1" : {
    			"type" : "string"
    		},
    		"attribute2" : {
    			"type" : "string"
    		},
          "myobjectbs" : {
            "type" : "object",
            "$ref" : "#/properties/myobjectbs"            
            }
          }
    	}
    }

    }
}

========================

The are 5 classes generated by this schema including (com.example.Myobjectb and com.example.Myobjectb_

AnotherObject.java use the list of Myobjectb_   while the top level has a list of Myobject

Should AnotherObject reuse the Myobjectb schema instead of generating a different class (only name is different)?

**Comment by joelittlejohn** [comment]
Yes, you're right, this both Test#myobjectbs and Test#AnotherObjects#myobjectbs should have the type List<Myobjectb>.

One solution to this is add `javaType` to myobjectb. It's a good idea to do this when you use refs.

**Comment by jshum2479** [comment]
The problem is we plan to share the schema with other teams and they want to use a different package names for all the schema,  the only way I can think of is to separate the myobjecta and myobjectb into different files and reference them in other array objects.

**Comment by ewernli** [comment]
We're having the same issue: 
* Types referenced several times are generated multiple times, postfixed with "_" 
* Types extended several times are generated multiple times, postfixed with "_Parent".

Is the best workaround to have one schema file per entity?

**Comment by ryeyao** [comment]
I'm having the same issue too. Is there any plan to fix this?

**Comment by vruc** [comment]
I think you should sepreate myobject as a new json and use $ref like these:
`{
  "id": "Test",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "myobjectas": {
      "type": "array",
      "items": {
        "name": "myobjecta",
        "type": "object",
        "$ref": "my-object.json#definitions/myObject"
      }
    },
    "myobjectbs": {
      "type": "array",
      "items": {
        "name": "myobjectb",
        "type": "object",
        "$ref": "my-object.json#definitions/myObject"
      }
    },
    "AnotherObjects": {
      "type": "object",
      "properties": {
        "attribute1": {
          "type": "string"
        },
        "attribute2": {
          "type": "string"
        },
        "myobjectbs": {
          "type": "object",
          "$ref": "#/properties/myobjectbs"
        }
      }
    }
  }
}`

`
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "myObject": {
      "type": "object",
      "properties": {
        "value": {
          "type": "string"
        }
      }
    }
  }
}
`
